contents/security/Https.md

```diff
@@ -0,0 +1,59 @@
+could redirect your traffic to monitored traffic by the attackers.
+
+All in all, the Internet architecture that we rely on for network transmission is very vulnerable to network sniffing. If we were to use http, which transmit packets in plain text, no confidentiality could be guranteed for our web application. Therefore, we need to use https as an end to end encryption to secure our network packets.
+
+## Why https is secure
+
+As aforementioned, our network is not secure, so how could https help? Https is built on top of http with addition of [SSL](https://www.digicert.com/ssl.htm) to encrypt the plain text message. The purpose of this encryption is to make sure only client and the server could decrypt the message with required keys, and sniffer cannot decrypt packets even though they may sniff packets. 
+
+There are mainly 3 encryption algorithms used in https, namely RSA, Diffie Hellman and Elliptic Curve Algorithm. They are more thoroughly explained in [Introduction to Cryptography](https://github.com/se-edu/learningresources "TODO: the link will be added after Dickson finishes on Cryptography") section. A simple understanding of why these algorithms prevent sniffers from decrypting the https packet by brute force is that the encryption process is of polynomial complexity and decryption by brute force is of exponential complexity. Therefore, when the encryption key size(number of bits in the key) is large enough(e.g. 2048 bits key for RSA), it becomes computationally impossible to decrypt the keys by brute force.
+
+Thus, by using https, we can be sure that even though our network packets are transmitted in an insecure network, sniffers cannot understand the content of our encrypted packets. 
+
+
+Besides providing secure network traffic, https also provides server validation through Certificate Authority(CA) architecture. A detailed explanation on CA is [here](https://www.globalsign.com/en-sg/ssl-information-center/what-are-certification-authorities-trust-hierarchies/) . In short, CA works by issuing the server a digital certificate that can only be produced by CA and client browsers can verify the server with digital certificate the server provides. 
+
+## How to set up https
+In order set up https on your server, you would need to have:
+1. A dedicated IP address for you server.
+2. Purchase a digital certificate from a CA.
+
+For example, one of the CA is [digicert](https://www.digicert.com) and they provide SSL [installation guide](https://www.digicert.com/ssl-certificate-installation.htm) for different platforms. Usually the SSL setup could be found on the CA that you purchased digital signature from.
```

Comment:
Might be good to mention the [Lets Encrypt](https://letsencrypt.org/) initiative. It provides free domain validation certificates, and automates configuration of common web servers.